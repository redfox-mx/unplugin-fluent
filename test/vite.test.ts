import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import { build } from 'vite'
import fluent from '../src/vite'
import { RollupOutput } from 'rollup'

async function getCode() {
  const bundle = (await build({
    plugins: [fluent()],
    build: {
      lib: {
        entry: resolve(__dirname, 'fixture/importer.js'),
        formats: ['es'],
      },
    },
  })) as RollupOutput[]

  const [output] = bundle
  return output.output.map((file) => {
    if (file.type === 'chunk') {
      return `//${file.fileName}\n${file.code}`
    } else {
      return file.fileName
    }
  }).join('\n')
}

describe('vite', () => {
  describe('fixtures', () => {
    for (const isProd of [false, true]) {
      it(`is prod environment ${isProd}`, async () => {
        process.env.NODE_ENV = isProd ? 'production' : 'development'

        const code = await getCode()
        expect(code).toMatchSnapshot()
      })
    }
  })
})
