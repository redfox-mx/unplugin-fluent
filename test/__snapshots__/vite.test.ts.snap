// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite > fixtures > is prod environment false 1`] = `
"//unplugin-fluent.js
const e = { body: [{ id: "hello-user", value: ["Hello, ", { type: "var", name: "userName" }, "!"], attributes: {} }, { id: "shared-photos", value: [{ type: "var", name: "userName" }, " ", { type: "select", selector: { type: "var", name: "photoCount" }, variants: [{ key: { type: "str", value: "one" }, value: "added a new photo" }, { key: { type: "str", value: "other" }, value: ["added ", { type: "var", name: "photoCount" }, " new photos"] }], star: 1 }, " to ", { type: "select", selector: { type: "var", name: "userGender" }, variants: [{ key: { type: "str", value: "male" }, value: "his stream" }, { key: { type: "str", value: "female" }, value: "her stream" }, { key: { type: "str", value: "other" }, value: "their stream" }], star: 2 }, "."], attributes: {} }] }, t = e;
export {
  t as output
};
"
`;

exports[`vite > fixtures > is prod environment true 1`] = `
"//unplugin-fluent.js
const e = { body: [{ id: "hello-user", value: ["Hello, ", { type: "var", name: "userName" }, "!"], attributes: {} }, { id: "shared-photos", value: [{ type: "var", name: "userName" }, " ", { type: "select", selector: { type: "var", name: "photoCount" }, variants: [{ key: { type: "str", value: "one" }, value: "added a new photo" }, { key: { type: "str", value: "other" }, value: ["added ", { type: "var", name: "photoCount" }, " new photos"] }], star: 1 }, " to ", { type: "select", selector: { type: "var", name: "userGender" }, variants: [{ key: { type: "str", value: "male" }, value: "his stream" }, { key: { type: "str", value: "female" }, value: "her stream" }, { key: { type: "str", value: "other" }, value: "their stream" }], star: 2 }, "."], attributes: {} }] }, t = e;
export {
  t as output
};
"
`;
